skbio/io/format/genbank.py

```diff
@@ -0,0 +1,468 @@
+        else:
+            metadata[header] = parsed
+    return sequence, metadata, positional_metadata
+
+
+def _parse_locus(lines):
+    '''Parse the line LOCUS.
+
+    Format:
+    #    Positions  Contents
+    #    ---------  --------
+    #    00:06      LOCUS
+    #    06:12      spaces
+    #    12:??      Locus name
+    #    ??:??      space
+    #    ??:29      Length of sequence, right-justified
+    #    29:33      space, bp/aa/rc, space
+    #    33:41      molecule type (can be blank): DNA, ssDNA, dsRNA, tRNA, etc.
+    #    41:42      space
+    #    42:51      Blank (implies linear), linear or circular
+    #    51:52      space
+    #    52:55      The division code (e.g. BCT, VRL, INV)
+    #    55:62      space
+    #    62:73      Date, in the form dd-MMM-yyyy (e.g., 15-MAR-1991)
+    '''
+    line = lines[0]
+    pattern = (r'LOCUS'
+               ' +([^\s]+)'
+               ' +([0-9]+)'
+               ' +(bp|aa|rc)'
+               ' +(.*DNA|.*RNA)?'
+               ' +(linear|circular)?'
+               ' +(PRI|ROD|MAM|VRT|INV|PLN|BCT|VRL|PHG|'
+               'SYN|UNA|EST|PAT|STS|GSS|HTG|HTC|ENV|CON)'
+               ' +([0-9]{2}-[A-Z]{3}-[0-9]{4})')
+    matches = re.match(pattern, line)
+    res = dict()
+    try:
+        res = dict(zip(
+            ['locus_name', 'size', 'unit',
+             'mol_type', 'shape',
+             'division', 'date'],
+            matches.groups()))
+    except:
+        raise GenbankFormatError(
+            "Could not parse the LOCUS line:\n%s" % line)
+
+    # if res['shape'] is None:
+    #     res['shape'] = 'linear'
+
+    res['size'] = int(res['size'])
+    res['date'] = datetime.strptime(res['date'], _TIME_FORMAT)
+    return res
+
+
+def _parse_reference(lines):
+    '''Parse single REFERENCE field.
+    '''
+    res = {}
+    section_splitter = yield_section(lambda x: not x.startswith(' ' * 11),
+                                     skip_blanks=True, strip=False)
+    for section in section_splitter(lines):
+        label, data = _parse_section_default(
+            section, join_delimitor=' ', return_label=True)
+        res[label] = data
+    return res
+
+
+def _parse_source(lines):
+    '''Parse SOURCE field.
+    '''
+    res = {}
+    section_splitter = yield_section(lambda x: not x.startswith(' ' * 11),
+                                     skip_blanks=True, strip=False)
+    # SOURCE line is not informative; skip it
+    _, organism = list(section_splitter(lines))
+
+    res['ORGANISM'] = organism[0].split(None, 1)[1].strip()
+    res['taxonomy'] = ' '.join([i.strip() for i in organism[1:]])
+    return res
+
+
+def _parse_features(lines, length):
+    '''Parse FEATURES field.
+    '''
+    features = []
+    positional_metadata = []
+    # skip the 1st FEATURES line
+    if lines[0].startswith('FEATURES'):
+        lines = lines[1:]
+    # magic number 20: the lines following header of each feature
+    # are at least indented with 20 spaces.
+    n = 20
```

Comment:
Actually it doesn't look like `n` is used again, so this would hopefully be more obvious: `feature_indent = " " * 20`